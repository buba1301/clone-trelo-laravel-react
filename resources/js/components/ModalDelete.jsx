import React from 'react';
import PropTypes from 'prop-types';

import { Button, Modal } from 'react-bootstrap';

const ModalDelete = ({
  showDeleteModal, handleClose, handleDelete, type,
}) => (
    <Modal show={showDeleteModal} onHide={handleClose}>
        <Modal.Header closeButton>
            <Modal.Title>Delete board</Modal.Title>
        </Modal.Header>
        <Modal.Body>
            Are you sure you want to delete this {type}?
        </Modal.Body>
        <Modal.Footer>
            <Button variant="secondary" onClick={handleClose}>
                Close
            </Button>
            <Button variant="danger" onClick={handleDelete}>
                Delete
            </Button>
        </Modal.Footer>
    </Modal>
);

export default ModalDelete;
