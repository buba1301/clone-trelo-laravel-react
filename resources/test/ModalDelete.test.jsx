/* eslint-disable max-len */
import React from 'react';
import { shallow, mount } from 'enzyme';
import ModalDelete from '../js/components/ModalDelete.jsx';

describe('Render DeleteModal', () => {
  const mockClose = jest.fn();
  const mockDelete = jest.fn();

  const props = {
    showDeleteModal: true,
    type: 'board',
    handleClose: mockClose,
    handleDelete: mockDelete,
  };

  const DeleteModalComponent = shallow(<ModalDelete { ...props } />);

  it('render correctly component', () => {
    expect(DeleteModalComponent).toMatchSnapshot();
  });

  it('render', () => {
    expect(DeleteModalComponent.find('Modal')).toHaveLength(1);
  });

  describe('render correct props', () => {
    const DeleteModalComponentForProps = mount(<ModalDelete { ...props } />); /// / почему пропсы есть только через mount
    it('check prop type', () => {
      expect(DeleteModalComponentForProps.props().showDeleteModal).toBeBoolean();
      expect(DeleteModalComponentForProps.props().type).toBeString();
      expect(DeleteModalComponentForProps.props().handleClose).toBeFunction();
      expect(DeleteModalComponentForProps.props().handleDelete).toBeFunction();
    });

    it('check prop value', () => {
      expect(DeleteModalComponentForProps.props().showDeleteModal).toBeTrue();
      expect(DeleteModalComponentForProps.props().type).toEqual('board');
    });
  });

  describe('when click close button', () => {
    it('call the props.handleClose', () => {
      DeleteModalComponent.find('#close').simulate('click', {
        preventDefault: () => {},
      });
      expect(mockClose).toHaveBeenCalledTimes(1);
    });
  });

  describe('when click delete button', () => {
    it('call the props.handleDelete', () => {
      DeleteModalComponent.find('#delete').simulate('click', {
        preventDefault: () => {},
      });
      expect(mockDelete).toHaveBeenCalledTimes(1);
    });
  });
});
