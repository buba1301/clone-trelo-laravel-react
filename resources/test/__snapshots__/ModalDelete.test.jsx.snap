// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DeleteModal render correctly component 1`] = `ShallowWrapper {}`;
